// react-semantify/lib/elements/flag.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var Unit = require("../commons/unit.js")(React);

  var defaultClassName = "flag";

  var Flag = React.createClass({
    displayName: "Flag",

    mixins: [ClassGenerator],

    render: function render() {
      var _props = this.props;
      var className = _props.className;
      var type = _props.type;
      var color = _props.color;

      var other = _objectWithoutProperties(_props, ["className", "type", "color"]);

      return React.createElement(Unit, _extends({}, other, {
        className: this.getClassName(defaultClassName),
        type: "icon",
        color: "null" }));
    }
  });

  return Flag;
};
