// react-semantify/lib/elements/segment.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var ColorSelector = require("../mixins/colorSelector.js")(React);
  var StateSelector = require("../mixins/stateSelector.js")(React);
  var Unit = require("../commons/unit.js")(React);

  var defaultClassName = "ui segment";

  var Segment = React.createClass({
    displayName: "Segment",

    mixins: [ClassGenerator, ColorSelector, StateSelector],

    render: function render() {
      var _props = this.props;
      var className = _props.className;
      var color = _props.color;

      var other = _objectWithoutProperties(_props, ["className", "color"]);

      return React.createElement(Unit, _extends({}, other, {
        className: this.getClassName(defaultClassName),
        type: "div",
        color: this.getColor(),
        disabled: this.getDisabled(),
        loading: this.getLoading() }));
    }
  });

  return Segment;
};
