// react-semantify/lib/elements/step.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var StateSelector = require("../mixins/stateSelector.js")(React);
  var Unit = require("../commons/unit.js")(React);

  var defaultClassName = "step";

  var Step = React.createClass({
    displayName: "Step",

    mixins: [ClassGenerator, StateSelector],

    render: function render() {
      var _props = this.props;
      var className = _props.className;

      var other = _objectWithoutProperties(_props, ["className"]);

      return React.createElement(
        Unit,
        _extends({}, other, {
          className: this.getClassName(defaultClassName),
          type: "div",
          color: "null",
          active: this.getActive(),
          completed: this.getCompleted(),
          disabled: this.getDisabled() }),
        this.props.children
      );
    }
  });

  return Step;
};
