// react-semantify/lib/mixins/stateSelector.js
"use strict";
module.exports = function (React) {

  var StateSelector = {

    propTypes: {
      disabled: React.PropTypes.bool,
      active: React.PropTypes.bool,
      loading: React.PropTypes.bool,
      focus: React.PropTypes.bool,
      error: React.PropTypes.bool,
      completed: React.PropTypes.bool,
      readOnly: React.PropTypes.bool,
      success: React.PropTypes.bool,
      warning: React.PropTypes.bool
    },

    getDisabled: function getDisabled() {
      var disabled = false;

      if (typeof this.props.disabled !== "undefined") {
        disabled = this.props.disabled;
      }

      return disabled;
    },

    getActive: function getActive() {
      var active = false;

      if (typeof this.props.active !== "undefined") {
        active = this.props.active;
      }

      return active;
    },

    getLoading: function getLoading() {
      var loading = false;

      if (typeof this.props.loading !== "undefined") {
        loading = this.props.loading;
      }

      return loading;
    },

    getFocus: function getFocus() {
      var focus = false;

      if (typeof this.props.focus !== "undefined") {
        focus = this.props.focus;
      }

      return focus;
    },

    getError: function getError() {
      var error = false;

      if (typeof this.props.error !== "undefined") {
        error = this.props.error;
      }

      return error;
    },

    getCompleted: function getCompleted() {
      var completed = false;

      if (typeof this.props.completed !== "undefined") {
        completed = this.props.completed;
      }

      return completed;
    },

    getReadOnly: function getReadOnly() {
      var readOnly = false;

      if (typeof this.props.readOnly !== "undefined") {
        readOnly = this.props.readOnly;
      }

      return readOnly;
    },

    getSuccess: function getSuccess() {
      var success = false;

      if (typeof this.props.success !== "undefined") {
        success = this.props.success;
      }

      return success;
    },

    getWarning: function getWarning() {
      var warning = false;

      if (typeof this.props.warning !== "undefined") {
        warning = this.props.warning;
      }

      return warning;
    }
  };

  return StateSelector;
};
