// react-semantify/lib/modules/dimmer.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var StateSelector = require("../mixins/stateSelector.js")(React);
  var Unit = require("../commons/unit.js")(React);

  var defaultClassName = "ui dimmer";

  var Dimmer = React.createClass({
    displayName: "Dimmer",

    mixins: [ClassGenerator, StateSelector],

    render: function render() {
      var _props = this.props;
      var className = _props.className;
      var color = _props.color;
      var type = _props.type;
      var disabled = _props.disabled;
      var active = _props.active;

      var other = _objectWithoutProperties(_props, ["className", "color", "type", "disabled", "active"]);

      return React.createElement(
        Unit,
        {
          className: this.getClassName(defaultClassName),
          color: "null",
          type: "div",
          disabled: this.getDisabled(),
          active: this.getActive() },
        this.props.children
      );
    },

    componentDidMount: function componentDidMount() {
      if (typeof this.props.init != "undefined") {
        if (this.props.init === false) {
          return;
        }

        if (this.props.init === true) {
          $(this.getDOMNode()).dimmer();
        } else {
          $(this.getDOMNode()).dimmer(this.props.init);
        }
      }
    }
  });

  return Dimmer;
};
