// react-semantify/lib/modules/dropdown.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var StateSelector = require("../mixins/stateSelector.js")(React);
  var Unit = require("../commons/unit.js")(React);

  var defaultClassName = "ui dropdown";

  var Dropdown = React.createClass({
    displayName: "Dropdown",

    mixins: [ClassGenerator, StateSelector],

    render: function render() {
      var _props = this.props;
      var className = _props.className;
      var color = _props.color;
      var type = _props.type;
      var error = _props.error;
      var disable = _props.disable;
      var active = _props.active;

      var other = _objectWithoutProperties(_props, ["className", "color", "type", "error", "disable", "active"]);

      if (this.getActive() || this.getDisabled()) {
        defaultClassName += " simple";
      }

      return React.createElement(
        Unit,
        _extends({}, other, {
          className: this.getClassName(defaultClassName),
          color: "null",
          type: "div",
          error: this.getError(),
          disable: this.getDisabled(),
          active: this.getActive() }),
        this.props.children
      );
    },

    componentDidMount: function componentDidMount() {
      if (typeof this.props.init != "undefined") {
        if (this.props.init === false) {
          return;
        }

        if (this.props.init === true) {
          $(this.getDOMNode()).dropdown();
        } else {
          $(this.getDOMNode()).dropdown(this.props.init);
        }
      }
    }
  });

  return Dropdown;
};
