// react-semantify/lib/modules/progress.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var StateSelector = require("../mixins/stateSelector.js")(React);

  var defaultClassName = "ui progress";

  var Progress = React.createClass({
    displayName: "Progress",

    mixins: [ClassGenerator, StateSelector],

    render: function render() {
      var _props = this.props;
      var className = _props.className;
      var percent = _props.percent;
      var value = _props.value;
      var total = _props.total;
      var active = _props.active;
      var success = _props.success;
      var warning = _props.warning;
      var error = _props.error;
      var disabled = _props.disabled;

      var other = _objectWithoutProperties(_props, ["className", "percent", "value", "total", "active", "success", "warning", "error", "disabled"]);

      var state = {
        active: this.getActive(),
        success: this.getSuccess(),
        warning: this.getWarning(),
        error: this.getError(),
        disabled: this.getDisabled()
      };

      return React.createElement(
        "div",
        _extends({}, other, {
          className: this.getClassName(defaultClassName, state),
          "data-percent": percent,
          "data-value": value,
          "data-total": total }),
        this.props.children
      );
    },

    componentDidMount: function componentDidMount() {
      if (typeof this.props.init != "undefined") {
        if (this.props.init === false) {
          return;
        }

        if (this.props.init === true) {
          $(this.getDOMNode()).progress();
        } else {
          $(this.getDOMNode()).progress(this.props.init);
        }
      }
    }
  });

  return Progress;
};
