// react-semantify/lib/modules/tab.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var StateSelector = require("../mixins/stateSelector.js")(React);

  var defaultClassName = "ui tab";

  var Tab = React.createClass({
    displayName: "Tab",

    mixins: [ClassGenerator, StateSelector],

    render: function render() {
      var _props = this.props;
      var className = _props.className;
      var active = _props.active;
      var loading = _props.loading;
      var tab = _props.tab;

      var other = _objectWithoutProperties(_props, ["className", "active", "loading", "tab"]);

      var state = {
        active: this.getActive(),
        loading: this.getLoading()
      };

      return React.createElement(
        "div",
        _extends({}, other, {
          className: this.getClassName(defaultClassName, state),
          "data-tab": tab }),
        this.props.children
      );
    },

    componentDidMount: function componentDidMount() {
      if (typeof this.props.init != "undefined") {
        if (this.props.init === false) {
          return;
        }

        if (this.props.init === true) {
          $(this.getDOMNode()).tab();
        } else {
          $(this.getDOMNode()).tab(this.props.init);
        }
      }
    }
  });

  return Tab;
};
