// react-semantify/lib/views/item.js
"use strict";

var _objectWithoutProperties = function (obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

module.exports = function (React) {

  var ClassGenerator = require("../mixins/classGenerator.js")(React);
  var TypeSelector = require("../mixins/typeSelector.js")(React);
  var Unit = require("../commons/unit.js")(React);

  var defaultClassName = "item";

  var Item = React.createClass({
    displayName: "Item",

    mixins: [ClassGenerator, TypeSelector],

    render: function render() {
      var _props = this.props;
      var className = _props.className;
      var type = _props.type;

      var other = _objectWithoutProperties(_props, ["className", "type"]);

      return React.createElement(
        Unit,
        _extends({}, other, {
          className: this.getClassName(defaultClassName),
          type: this.getType(),
          color: "null",
          value: this.props.value }),
        this.props.children
      );
    }
  });

  return Item;
};
